src/workloads/execution/PipelineUpdate.yml

```diff
@@ -0,0 +1,352 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+
+Actors:
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: &db test
+    Threads: 1
+    CollectionCount: 1
+    DocumentCount: 10000
+    BatchSize: 2000
+    Document: &smallDocument    # Around 200B size.
+      t: &partitionIDSmall {^RandomInt: {min: 0, max: 1000}}
+      largeStr: &mediumString {^FastRandomString: {length: 128}}
+      int: &integer {^RandomInt: {min: 0, max: 100}}
+      str: &string {^RandomString: {length: {^RandomInt: {min: 10, max: 20}}}}
+      obj: &object {a: *integer, b: *integer, c: *string}
+      nestedObj: &nestedObject {
+        int: *integer,
+        str: *string,
+        arr: &array [*integer, *integer, *integer],
+        nestedArr: [*integer, *string, *object, *array]}
+    Indexes:
+    - keys: {t: 1}
+  - Repeat: 1
+    Database: *db
+    Threads: 1
+    CollectionCount: 1
+    DocumentCount: 10000
+    BatchSize: 2000
+    Document: &mediumDocument   # Around 1KB size.
+      t: &partitionIDMedium {^RandomInt: {min: 1001, max: 2000}}
+      largeStr: &largeString {^FastRandomString: {length: 1024}}
+      int: *integer
+      str: *string
+      obj: *object
+      nestedObj: *nestedObject
+    Indexes:
+    - keys: {t: 1}
+  - Repeat: 1
+    Database: *db
+    Threads: 1
+    CollectionCount: 1
+    DocumentCount: 10000
+    BatchSize: 2000
+    Document: &largeDocument    # Around 100KB Size.
+      t: &partitionIDLarge {^RandomInt: {min: 2001, max: 3000}}
+      int: *integer
+      str: *string
+      largeStr: &giantString {^FastRandomString: {length: 100000}}
+      obj: *object
+      nestedObj: *nestedObject
+    Indexes:
+    - keys: {t: 1}
+  - &Nop {Nop: true}
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: RunUpdates
+  Type: RunCommand
+  Threads: 8
```

Comment:
Only tried 4 and 8 threads, haven't well tuned this workload yet. Will do it once the workload is set.

Reply:
Sounds good.